worker/caasapplicationprovisioner/application.go

```diff
@@ -211,32 +282,58 @@ func (a *appWorker) loop() error {
 		return nil
 	}
 
+	var scaleChan <-chan time.Time
+	var scaleTries int
+	var trustChan <-chan time.Time
+	var trustTries int
+	const maxRetries = 10
+	const retryDelay = 3 * time.Second
```

Comment:
Lets use github.com/juju/utils/v2 AttemptStrategy

Reply:
I wanted to use something like this, but couldn't figure out a non-messy way: how would you use `AttemptStrategy` here, given the context of a big non-blocking select with a delay between each attempt?